pyvista/plotting/widgets.py

```diff
@@ -1730,6 +1732,49 @@ def _the_callback(widget, event):
         self.button_widgets.append(button_widget)
         return button_widget
 
+    def add_camera_orientation_widget(self, animate=True, n_frames=20):
+        """Add a camera orientation widget to the active renderer.
+
+        .. note::
+           This widget requires ``vtk>=9.1.0``.
+
+        Parameters
+        ----------
+        animate : bool, optional
+            Enable or disable jump-to-axis-view animation.
+        n_frames : int, optional
+            The number of frames to animate the jump-to-axis-viewpoint feature.
+
+        Examples
+        --------
+        Add a camera orientation widget to the scene.
+
+        >>> import pyvista
+        >>> mesh = pyvista.Cube()
+        >>> plotter = pyvista.Plotter()
+        >>> plotter.add_mesh(mesh, scalars=range(6), show_scalar_bar=False)
+        >>> plotter.add_camera_orientation_widget()
+        >>> plotter.show()
+
+        """
+        try:
+            from vtkmodules.vtkInteractionWidgets import vtkCameraOrientationWidget
```

Comment:
This no longer follows the usage of `_vtk` is there any way to generalize this as I also have filters that I'd like to add from 9.1.0?  Unless this is generalized, all new features will be ad hoc like this.

Reply:
There's (at least) three ways of doing this:

1. Leave as is. It's adhoc, but means we don't need to keep our functionality in `_vtk`. Probably not the best approach

2. We could create an empty placeholder for VTK 9.1.0 compatibility as in
https://github.com/pyvista/pyvista/blob/363bca639c9050b2be5832e5105687691c43ea45/pyvista/_vtk.py#L481-L487
and then import always.

3.  Attempt to import and pass.

I think (2) will probably be the most robust.

---
Side note: maintaining backwards compatibility with vtk within pyvista has always been a goal of the project, since requiring a newer version of VTK has always been a non-goal since we want pyvista to be widely available. This is hard to do and we should try to do it right.

Reply:
In that case we could have a `_vtk_compat.py` or similar, where all these placeholders are. In `_vtk` we can do
```py
try:
    from vtk.vtkSuperSpecificSubmodule import vtkUsefulFunctionality
except ImportError:
    from ._vtk_compat import vtkUsefulFunctionality
```

Any obvious problem with this approach?

Reply:
> Side note: maintaining backwards compatibility with vtk within pyvista has always been a goal of the project, since requiring a newer version of VTK has always been a non-goal since we want pyvista to be widely available. This is hard to do and we should try to do it right.

Right now, I think we are only testing with the vtk==8.2 and latest vtk.  We need to add back vtk==9.0.x.

I like @adeak suggestion, but I would generalize even further:

```py
from ._vtk_compat import vtkVersionError

try:
    from vtk.vtkSuperSpecificSubmodule import vtkUsefulFunctionality
except ImportError:
    vtkUsefulFunctionality = vtkVersionError("vtkUsefulFunctionality only supported for vtk versions >=9.0.0")
    
``` 

Then we only need 1 placeholder for all functions (or maybe some small number for different kinds).

Reply:
>  We need to add back vtk==9.0.x.
+1

>  I would generalize even further:

Good point. If we end up having too many of these we'll have to resist the temptation to do this dynamically in a loop :P

Reply:
Can we implement this in a follow up PR? I wanted to limit the scope of the PR just to adding the widget, but it seems like whenever I implement a feature there are about 10 issues I uncover (probably my own doing...).

Reply:
> 

I'm fine with this approach.  I wanted to highlight it since this is the first PR with features from 9.1.0.